### PR TITLE
Add EmbeddedSource baseline tests and compound extension mapping fix

### LINKED ISSUE #130: TDD: add baseline tests for AgentSandbox.Core.Importing.EmbeddedSource
## Summary

Coverage hotspot: `AgentSandbox.Core/Importing/EmbeddedSource.cs` (0% line coverage).

## Goal

Establish baseline unit tests for embedded skill source loading behavior and edge conditions.

## Scope

- Add tests for discovery/loading success paths.
- Add tests for missing/invalid embedded resources and parse failures.
- Verify stable contracts for returned metadata/content.

## Acceptance Criteria

- `EmbeddedSource.cs` has meaningful baseline test coverage.
- Edge/error paths are covered with explicit assertions.
- Full test suite remains green.

### PR DESCRIPTION
**Pull request by hahahahahaiyiwen**
Closes #130

## Summary
- add baseline EmbeddedSource tests using embedded assembly resources for success, missing-prefix, and invalid-skill paths
- fix EmbeddedSource path conversion to preserve common compound filename extensions (e.g. .tar.gz, .d.ts, .min.js, .d.ts.map)
- add embedded test fixtures and project logical-name mappings for deterministic resource loading

## Verification
- dotnet test tests/AgentSandbox.Tests/AgentSandbox.Tests.csproj --filter "FullyQualifiedName~EmbeddedSourceTests"
- dotnet test AgentSandbox.sln

### CODE REVIEW

File: tests/AgentSandbox.Tests/EmbeddedSourceTests.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    [Fact]
+    public void GetFiles_PrefixWithTrailingDot_LoadsSameResources()
+    {
+        var source = new EmbeddedSource(typeof(EmbeddedSourceTests).Assembly, ValidPrefix + ".");
+
+        var files = source.GetFiles().ToList();
+
+        Assert.Equal(5, files.Count);
```
[the remark this answers is not in the document]
Comment (hahahahahaiyiwen) [reply tests/AgentSandbox.Tests/EmbeddedSourceTests.cs]: @copilot-pull-request-reviewer: Addressed. The missing fixtures have now been committed and pushed (203c5bb), and the exact-count/path assertions now match the embedded resources present in the branch. Verified with `dotnet test ... --filter FullyQualifiedName~EmbeddedSourceTests` and full `dotnet test AgentSandbox.sln`.
